src/rom_utils.cpp

```diff
@@ -137,34 +223,41 @@ bool load_rom(emu_shared_context_t *ctx, const char *filename) {
         std::cerr << "Could not read all of ROM image." << std::endl;
         free(full_image);
         stream.close();
-        return false;
+        return ROM_FAIL_READ;
     }
 
     stream.close();
 
-    //assume file is legitimate ROM
+    uint8_t load_rom_rc = validate_rom(full_image, rom_size);
+
+
+    if (load_rom_rc != ROM_OK) { // Anything other than OK is an error
+        if (load_rom_rc == ROM_FAIL_FULL_VALIDATION); // This is just a warning, skip this case
```

Comment:
This works, but why the empty if block?

Reply:
OK so I get that this looks suspect but it made more sense to me to catch anything that wasn't ROM_OK and then just filter out the non-fatal error rather than a big long if statement.

I guess I figured if there was more to add in the future, it's just one line to add per new case.

Reply:
The ROM_OK check is fine, but it's probably better to invert the condition on the inner block for now. If there are more non-fatal codes later we can change it back.

Reply:
👌